lib/commands/project.ts

```diff
@@ -28,25 +34,108 @@ export class ProjectCommandBase implements ICommand {
 	}
 }
 
+export class FileDescriptor {
+	constructor(public path: string, public type: string) { }
+}
+
+export class InitProjectCommandBase extends ProjectCommandBase {
+	private cordovaFiles: FileDescriptor[];
+	protected projectDir: string;
+	protected tnsModulesDir: FileDescriptor;
+	protected bootstrapFile: FileDescriptor;
+	protected indexHtml: FileDescriptor;
+	protected projectFilesDescriptors: any;
+
+	constructor(protected $project: Project.IProject,
+				protected $errors: IErrors,
+				protected $fs: IFileSystem,
+				protected $logger: ILogger,
+				protected $projectNameValidator?: IProjectNameValidator) {
+		super($project, $errors, $projectNameValidator);
+
+		this.projectDir = this.$project.getNewProjectDir();
+		this.tnsModulesDir = new FileDescriptor(path.join(this.projectDir, "tns_modules"), "directory");
+		this.bootstrapFile = new FileDescriptor(path.join(this.projectDir, "app", "bootstrap.js"), "file");
+		this.indexHtml = new FileDescriptor(path.join(this.projectDir, "index.html"), "file");
+		this.cordovaFiles = _.map(Object.keys(MobileHelper.platformCapabilities), platform => {
+			return new FileDescriptor(util.format("cordova.%s.js", platform).toLowerCase(), "file");
+		});
+
+		this.generateMandatoryAndForbiddenFiles();
+	}
+
+	protected isProjectType(projectType: string): IFuture<boolean> {
+		return (() => {
+			var result = true;
+			var projectData = this.projectFilesDescriptors[projectType];
+
+			_.each(projectData.mandatoryFiles, (file: FileDescriptor) => {
+				if(!this.$fs.exists(file.path).wait()) {
+					this.$logger.trace("Missing %s %s. This is not a %s project.", file.path, file.type, projectType);
+					result = false;
+				}
+			});
+
+			_.each(projectData.forbiddenFiles, (file: FileDescriptor) => {
+				if(this.$fs.exists(file.path).wait()) {
+					this.$logger.trace("Found %s %s. This is not a %s project.", file.path, file.type, projectType);
```

Comment:
Wonderful about the double %s.

To avoid the a/an %s controversy, we can go with any of the following:

Found %s %s, typically not associated with %s projects.
Found %s %s. This is not typically associated with %s projects.
Found %s %s. Typically, %s projects do not contain %s.
Found %s %s. Typically, %s does not exist in %s projects.
Found %s %s. Cannot complete %s project initialization.
Found %s %s. The project type is not %s. (or make this The project type is NOT %s.)

My personal favorites are the last two.